src/commands/shell.ts

```diff
@@ -0,0 +1,120 @@
+import { Command } from "../command";
+import { start } from "repl";
+import { yellow, red, bold } from "cli-color";
+import * as admin from "firebase-admin";
+import { getFirebaseConfig } from "../functionsConfig";
+import { getAccessToken } from "../api";
+import { getRefreshToken } from "../auth";
+import { requireAuth } from "../requireAuth";
+import { FirebaseError } from "../error";
+import { createContext, runInContext, Context, Script } from "vm";
+import { readFileSync, existsSync } from "fs";
+import fetch, { Response } from "node-fetch";
+import { DocumentSnapshot, Firestore, QuerySnapshot } from "@google-cloud/firestore";
+import api = require("../api");
+
+async function runScript(sandbox: Context, scriptPath: string) {
+  if (!existsSync(scriptPath)) {
+    throw new FirebaseError(`Script file ${bold(scriptPath)} does not exist.`);
+  }
+
+  try {
+    const script = new Script(readFileSync(scriptPath, { encoding: "utf8" }), {
+      filename: scriptPath,
+    });
+    return await Promise.resolve(script.runInContext(sandbox));
```

Comment:
Why the additional wrapping? Also, does it return something worthwhile? Maybe just `await` the script and remove the return.

```suggestion
    return await script.runInContext(sandbox);
```

(I'm seeing a comment regarding the wrapping below - if it's the same condition, let's add the same comment here)

Reply:
Done.